console/src/config.rs

```diff
@@ -176,3 +191,13 @@ fn parse_true_color(s: &str) -> bool {
     let s = s.trim();
     s.eq_ignore_ascii_case("truecolor") || s.eq_ignore_ascii_case("24bit")
 }
+
+fn parse_retain_for(s: &str) -> Result<RetainFor, String> {
+    match s {
+        "none" => Ok(RetainFor(None)),
+        _ => match s.parse::<humantime::Duration>() {
+            Ok(duration) => Ok(RetainFor(Some(duration.into()))),
+            Err(error) => Err(error.to_string()),
+        },
+    }
+}
```

Comment:
nit/TIOLI: this could be an implementation of the `FromStr` trait for `RetainFor`, since `RetainFor` is a type we define in this crate. then, we wouldn't need the `parse(try_from_str = parse_retain_for))` in the `clap` attribute.

not a blocker